src/plugins/accounting.cpp

```diff
@@ -35,3 +35,67 @@ Association* get_association (int userid,
 
     return &bank_it->second;
 }
+
+
+std::string Association::to_json () const
+{
+    json_t *u = json_object ();
+
+    json_object_set_new (u, "bank_name", json_string (bank_name.c_str ()));
+    json_object_set_new (u, "fairshare", json_real (fairshare));
+    json_object_set_new (u, "max_run_jobs", json_integer (max_run_jobs));
+    json_object_set_new (u, "cur_run_jobs", json_integer (cur_run_jobs));
+    json_object_set_new (u, "max_active_jobs", json_integer (max_active_jobs));
+    json_object_set_new (u, "cur_active_jobs", json_integer (cur_active_jobs));
+
+    json_t *held_job_ids = json_array ();
+    for (const auto &job_id : held_jobs) {
+        json_array_append_new (held_job_ids, json_integer (job_id));
+    }
+    json_object_set_new (u, "held_jobs", held_job_ids);
+
+    json_t *user_queues = json_array ();
+    for (const auto &queue : queues) {
+        json_array_append_new (user_queues, json_string (queue.c_str ()));
+    }
+    json_object_set_new (u, "queues", user_queues);
+
+    json_object_set_new (u, "queue_factor", json_integer (queue_factor));
+    json_object_set_new (u, "active", json_integer (active));
+
+    char *json_str = json_dumps (u, JSON_INDENT (4));
+    std::string result (json_str);
+    free (json_str);
+    json_decref (u);
+
+    return result;
+}
+
+
+json_t* convert_map_to_json (std::map<int, std::map<std::string, Association>>
+                                &users)
+{
+    json_t *accounting_data = json_array ();
+
+    // each Association in the users map is a pair; the first item is the
+    // userid and the second is a list of banks they belong to
+    for (const auto& association : users) {
+        json_t *u = json_object ();
+        json_object_set_new (u, "userid", json_integer (association.first));
+
+        // the user might belong to multiple banks, so we need to iterate
+        // through each one
+        json_t *banks = json_array ();
+        for (const auto& bank : association.second) {
+            // bank.second refers to an Association object
+            Association a = bank.second;
+            json_t *b = json_loads (a.to_json ().c_str (), 0, nullptr);
```

Comment:
To avoid writing it out to a string and reading it back in again, I wonder if you could have an `Association` method like `to_json_t` that returns a `json_t` and then have the `to_json` method invoke that method before turning it into a string?